test/Resolver/ServiceTest.php

```diff
@@ -0,0 +1,138 @@
+
+use Magento\Upward\Definition;
+use Magento\Upward\DefinitionIterator;
+use Magento\Upward\Resolver\Service;
+use Mockery;
+use Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+use PHPUnit\Framework\TestCase;
+use Zend\Http\Client;
+use Zend\Http\Response;
+use function BeBat\Verify\verify;
+
+class ServiceTest extends TestCase
+{
+    use MockeryPHPUnitIntegration;
+
+    /**
+     * @var DefinitionIterator|Mockery\MockInterface
+     */
+    private $definitionIteratorMock;
+
+    /**
+     * @var Service
+     */
+    private $resolver;
+
+    protected function setUp(): void
+    {
+        $this->resolver               = new Service();
+        $this->definitionIteratorMock = Mockery::mock(DefinitionIterator::class);
+        $this->resolver->setIterator($this->definitionIteratorMock);
+    }
+
+    public function testIndicator(): void
+    {
+        verify($this->resolver->getIndicator())->is()->sameAs('url');
+    }
+
+    public function testIsValid(): void
+    {
+        $validDefinition           = new Definition(['url' => '/graphql', 'query' => 'gql']);
+        $validWithMethodDefinition = new Definition(['url' => '/graphql', 'query' => 'gql', 'method' => 'GET']);
+        $invalidNoURL              = new Definition(['query' => 'gql']);
+        $invalidNoQuery            = new Definition(['url' => '/graphql']);
+        $invalidUnsupportedMethod  = new Definition(['url' => '/graphql', 'query' => 'gql', 'method' => 'PUT']);
+
+        $this->definitionIteratorMock->shouldReceive('get')
+            ->twice()
+            ->with('method', Mockery::type(Definition::class))
+            ->andReturnUsing(function (string $key, Definition $definition) {
+                return $definition->get($key);
+            });
+
+        verify($this->resolver->isValid($validDefinition))->is()->true();
+        verify($this->resolver->isValid($validWithMethodDefinition))->is()->true();
+        verify($this->resolver->isValid($invalidNoURL))->is()->false();
+        verify($this->resolver->isValid($invalidNoQuery))->is()->false();
+        verify($this->resolver->isValid($invalidUnsupportedMethod))->is()->false();
+    }
+
+    public function testResolve(): void
+    {
+        $definition            = new Definition(['url' => '/graphql', 'query' => 'gql']);
+        $expectedRequestBody   = json_encode(['query' => 'gql', 'variables' => []]);
+        $expectedResponseArray = ['data' => ['key' => 'value']];
+
+        $this->definitionIteratorMock->shouldReceive('get')
+            ->twice()
+            ->with(Mockery::type('string'), Mockery::type(Definition::class))
+            ->andReturnUsing(function (string $key, Definition $definition) {
+                return $definition->get($key);
+            });
+
+        $responseMock = Mockery::mock(Response::class);
+        $responseMock->shouldReceive('getBody')->andReturn('{"data":{"key":"value"}}');
+
+        $zendClientMock = Mockery::mock('overload:' . Client::class);
```

Comment:
Whenever we do an `overload` mock, it's probably safest to put it in a separate process, we should add either `@runInSeparateProcess` to this test and `testResolveWithConfiguration` or add `@runTestsInSeparateProcesses` to this class.

Reply:
Added test isolation (also to `ProxyTest` that overloaded and missed this annotation as well)